source/plugin/basics/data/datamanipulator.rst

```diff
@@ -0,0 +1,110 @@
+    Optional<DataManipulatorBuilder<TreeData>> builder = registry.getBuilder(TreeData.class);
+
+**Example: Creating TreeData objects using the builder**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataHolder;
+    import org.spongepowered.api.data.DataManipulatorBuilder;
+    import org.spongepowered.api.data.manipulator.block.TreeData;
+
+    TreeData defaultData = builder.get().create();
+
+    DataHolder holder = ...;
+    Optional<TreeData> copiedData = builder.get().createFrom(holder);
+
+Manipulating Data
+=================
+
+Depending on its subclass, a data holder provides methods for accessing and modifying the contained data. Refer to the JavaDocs of the respective data manipulator class for further information. However, some common behaviours are explained below.
+
+Boolean Data
+~~~~~~~~~~~~
+
+Data manipulators representing a single boolean value do not contain any methods for altering their data. If such a data manipulator can be built from a data holder, it means that the data represented evaluates to ``true``. If it cannot be built but is compatible to the data holder, it evaluates to ``false``.
+
+An example for this is ``Wetdata``. A data holder is wet, if and only if a ``WetData`` object can be created from it.
+
+**Example: Check if a DataHolder is wet**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataHolder;
+    import org.spongepowered.api.data.DataManipulatorBuilder;
+    import org.spongepowered.api.data.manipulator.WetData;
+
+    DataManipulatorBuilder<WetData> builder = ...;
+    DataHolder holder = ...;
+    boolean isWet = builder.createFrom(holder).isPresent();
+
+
+Validity Checks
+~~~~~~~~~~~~~~~
+
+For some data manipulators, not every value that can be supplied will also be valid. Therefore, data manipulators may reject invalid values. There are currently two possibilities how invalid values may be handled.
+
+The common way of reacting to invalid input is throwing an ``IllegalArgumentException``. This is also what generally will occur on data manipulators. Some data manipulators will also provide means to check if a value is valid. Refer to the API docs of your respective data manipulator to learn which values are valid and if it is possible to perform a validity check before trying to set a value.
+
+However, if a data manipulator accepts data in key-value pairings (``Map``\ s) it may occur that only a portion of the supplied data is invalid and other portions can be applied. Since the operation of setting mapped data is not guaranteed to be atomic, it may be possible that the data was partially applied. The returned ``DataTransactionResult`` allows for checking if the operation was successful and also specifies which data was accepted. A more detailed take on ``DataTransactionResult`` can be found :ref:`here <dataapi-transactionresult>`. This validity check may be bypassed using the ``setUnsafe`` methods.
```

Comment:
is setUnsafe only applicable to mapped data, or is it applicable to any data where it can be found?

If it's only applicable to mapped data, should it be available to other data types?